githubstat/metrics_commit_request.go

```diff
@@ -0,0 +1,123 @@
+package githubstat
+
+import (
+	"fmt"
+	"github.com/google/go-github/github"
+	"strings"
+	"time"
+)
+
+type WarpRepositoryCommit struct {
+	RepositoryCommit *github.RepositoryCommit
+	Owner            string
+	Repo             string
+	MergedAt         *time.Time
+}
+
+// sameCommitter means the pr author was same with the commit author
+func (m *WarpRepositoryCommit) findMergedTime(client *github.Client, sameCommitter bool) bool {
+	SHA := m.RepositoryCommit.SHA
+	owner := m.Owner
+	repo := m.Repo
+	author := m.RepositoryCommit.Author.Login
+	if !sameCommitter {
+		author = m.RepositoryCommit.Committer.Login
+	}
+	// TODO consider using template
+	query := *SHA + " repo:" + owner + "/" + repo + " type:pr" + " author:" + *author
+	pr := findPRfromCommit(client, query)
+	if pr == nil {
+		return false
+	}
+	m.MergedAt = pr.ClosedAt
```

Comment:
because there is no `MergeAt` attr, only `ClosedAt`, `CreatedAt` and `UpdateAt` attributes in the issue(pull request) struct, `ClosedAt` is the closest time stamp to the merge time